tiff-2-png-converter.py

```diff
@@ -7,100 +7,249 @@
+    max_data = np.amax(np.amax(img_3d_array, axis=2))
+    logging.info("Minimum value in the whole array:%d" % (min_data))
+    logging.info("Maximum value in the whole array:%d" % (max_data))
+    min_max_scale = np.arange(min_data, max_data)
+
+
+def merge_rgb(image_channels, rgb_images, width, height, bits_per_sample):
+    zero_image = np.zeros((width, height), dtype=bits_per_sample)
+    with typer.progressbar(
+        range(arg_values["nimgs"]), label="Mergin channels"
+    ) as progress:
+        for z in progress:
+            r_channel = np.zeros((width, height), dtype=bits_per_sample)
+            g_channel = np.zeros((width, height), dtype=bits_per_sample)
+            b_channel = np.zeros((width, height), dtype=bits_per_sample)
+            current_image_channels = [r_channel, g_channel, b_channel]
+            ## loop over the channels :  r = 0, g = 1, b =2
+            for i in range(3):
+                current_channel = image_channels[i]
+                if current_channel[z] is not None:
+                    current_image_channels[i] = current_channel[z]
+                    if arg_values["equalize_histogram"]:
+                        current_image_channels[i] = arg_values[
+                            "equalize_histogram_method"
+                        ](current_image_channels[i])
+                else:
+                    current_image_channels[i] = zero_image
+
+            merged_image = cv2.merge(current_image_channels)
 
-        if args.gamma:
-            TiffProcessor.gamma = True
-            TiffProcessor.gamma_val = args.gamma
-        if args.num_imgs:
-            TiffProcessor.num_imgs = int(args.num_imgs)
-
-    def convert_to_png(self):
-        list_files, width, heigth = self.list_folder_files(self.source)
-        if self.num_imgs is None:
-            self.num_images = len(list_files)
-        image_r = [None] * self.num_imgs
-        image_g = [None] * self.num_imgs
-        image_b = [None] * self.num_imgs
-        images = [None] * self.num_imgs
-        bits_per_sample = ""
-        print_progress_bar(
-            0, self.num_imgs, prefix="grouping channels:", suffix="Complete", length=50
-        )
-        for i in range(len(list_files)):
+            if arg_values["gamma"] is not None:
+                float_gamma = float(arg_values["gamma"])
+                merged_image = iph.gamma_correction(merged_image, float_gamma)
+            rgb_images[z] = merged_image
+
+
+def build_image_sequence(volume, width, height, dtype):
+    z_slices = arg_values["nimgs"] + 1
+    dim = math.ceil(math.sqrt(z_slices))
+    max_res = math.floor(4096 / dim)
+    downscale = math.ceil(max(width, height) / max_res)
+    if arg_values["verbose"]:
+        logging.info("resolution_image_before " + str(width) + " " + str(height))
+        logging.info("downscale " + str(downscale))
+
+    resolution_image = (int(width / downscale), int(height / downscale))
+    count = 0
+    image_out = np.zeros(())
```

Comment:
You have the size of the image, it'll be more efficient if you pre-allocate the array.